cpp/oneapi/dal/algo/knn/backend/cpu/infer_kernel_brute_force.cpp

```diff
@@ -98,14 +99,34 @@ static infer_result<Task> call_daal_kernel(const context_cpu &ctx,
     const auto daal_voting_mode = convert_to_daal_bf_voting_mode(desc.get_voting_mode());
     daal_parameter.voteWeights = daal_voting_mode;
 
-    interop::status_to_exception(interop::call_daal_kernel<Float, daal_knn_bf_kernel_t>(
-        ctx,
-        daal_data.get(),
-        dal::detail::get_impl(m).get_interop()->get_daal_model().get(),
-        daal_labels.get(),
-        daal_indices.get(),
-        daal_distance.get(),
-        &daal_parameter));
+    if (!dynamic_cast<brute_force_model_impl<Task> *>(&dal::detail::get_impl(m))) {
+        throw internal_error{ dal::detail::error_messages::unsupported_knn_model() };
+    }
+
+    const auto deserialized_model =
+        dynamic_cast<brute_force_model_impl<Task> *>(&dal::detail::get_impl(m));
+
+    const auto daal_train_data = interop::convert_to_daal_table<Float>(deserialized_model->data_);
+    const auto daal_train_labels =
+        interop::convert_to_daal_table<Float>(deserialized_model->labels_);
+    const std::int64_t column_count = daal_train_data->getNumberOfColumns();
+
+    Status status;
+    const auto model_ptr = daal_knn::ModelPtr(new daal_knn::Model(column_count));
+    interop::status_to_exception(status);
```

Comment:
Why do you need check status?

Reply:
Copy code from train stage. Do we need to remove this check in both places?

Reply:
sure, this is unused variable :)

Reply:
removed from all files